tutor/templates/build/openedx/Dockerfile

```diff
@@ -144,38 +145,53 @@ RUN mkdir -p "$STATIC_ROOT_LMS"
 
 
 ###########################################################################################################################
-############## BUNDLES
+############## FRONTEND SOURCES
 ###########################################################################################################################
 
-FROM frontend-requirements as bundles
+# Intermediate stage to capture *only* frontend source files.
+# Frontend sources don't change very often, so this allows Docker to skip Webpack and Sass
+# compilation in most warm-cache situations.
+FROM frontend-requirements as frontend-sources
+
+# Copy in all edx-platform source dirs.
+COPY --link --from=edx-platform /lms lms
+COPY --link --from=edx-platform /cms cms
+COPY --link --from=edx-platform /common common
+COPY --link --from=edx-platform /openedx openedx
+COPY --link --from=edx-platform /xmodule xmodule
+
+# Now, delete everything that isn't relevant to the Webpack or Sass builds.
+# We expect that these patterns capture all the relevant static assets, although it is possible
+# that an upstream change could break that expectation. However, new frontends are generally
+# added to MFEs instead of edx-platform, so we don't expect much churn if any.
+RUN find lms cms common openedx xmodule -type f -a \! \
+	\( -path '*/static/*' \
+	-o -path '*/assets/*' \
+	-o -path '*/js/*' \
+	-o -name '*.underscore' \
+	\) -delete
+
+# These dirs are generated by `npm run postinstall`, so they may be missing from a local
+# edx-platform bind-mount, so the COPY --from=edx-platform above may have clobbered them
+# away. So, we must copy them back in from frontend-requirements.
+COPY --link --from=frontend-requirements /openedx/edx-platform/common/static/common/js/vendor common/static/common/js/vendor
+COPY --link --from=frontend-requirements /openedx/edx-platform/common/static/common/css/vendor common/static/common/css/vendor
+
+# Finally, clear away any dirs that are now empty.
+RUN find . -type d -empty -delete
 
+
+###########################################################################################################################
+############## WEBPACK BUNDLES
+###########################################################################################################################
+
+FROM frontend-sources as bundles
 COPY --link --from=edx-platform /.babelrc .babelrc
-COPY --link --from=edx-platform /webpack-config/file-lists.js webpack-config/file-lists.js
+COPY --link --from=edx-platform /webpack-config webpack-config
 COPY --link --from=edx-platform /webpack.builtinblocks.config.js webpack.builtinblocks.config.js
 COPY --link --from=edx-platform /webpack.common.config.js webpack.common.config.js
 COPY --link --from=edx-platform /webpack.dev.config.js webpack.dev.config.js
 COPY --link --from=edx-platform /webpack.prod.config.js webpack.prod.config.js
```

Comment:
Unfortunately, we still must copy in the Webpack configs separately. This could be improved upstream by shoving all of these webpack configs into the /webpack-config directory, and then just copying that one directory in.